third-party/realdds/src/dds-device-server.cpp

```diff
@@ -379,45 +379,39 @@ void dds_device_server::handle_query_option( const nlohmann::json & j, nlohmann:
     }
     else
     {
-        if( owner_name.empty() )
-            owner_name = "device";
+        if( stream_name.empty() )
+            stream_name = "device";
         else
-            owner_name = "'" + owner_name + "'";
-        DDS_THROW( runtime_error, owner_name + " option '" + option_name + "' not found" );
+            stream_name = "'" + stream_name + "'";
+        DDS_THROW( runtime_error, stream_name + " option '" + option_name + "' not found" );
     }
 }
 
 
 std::shared_ptr< dds_option > realdds::dds_device_server::find_option( const std::string & option_name,
-                                                                       const std::string & owner_name )
+                                                                       const std::string & stream_name ) const
 {
-    if( owner_name.empty() )
+    if( stream_name.empty() )
     {
         for( auto & option : _options )
         {
             if( option->get_name() == option_name )
-            {
                 return option;
-            }
         }
     }
     else
     {
         // Find option in owner stream
-        for( auto & stream_it : _stream_name_to_server )
+        auto stream_it = _stream_name_to_server.find( stream_name );
+        if( stream_it != _stream_name_to_server.end() )
         {
-            if( stream_it.first == owner_name )
+            for( auto & option : stream_it->second->options() )
             {
-                for( auto & option : stream_it.second->options() )
-                {
-                    if( option->get_name() == option_name )
-                    {
-                        return option;
-                    }
-                }
+                if( option->get_name() == option_name )
+                    return option;
             }
         }
     }
 
-    return nullptr;
+    return {};
```

Comment:
I don't understand this change.
It creates an empty shared_ptr? shared_ptr to default initialized dds_option?
Why is this better then returning nullptr?

Reply:
`return {}` simply says "return an object constructed with the default state". The default state (just as if you wrote `shared_ptr<T>()` or `shared_ptr<T>{}` is an "empty" shared_ptr is a shared_ptr with a nullptr. Constructing with `nullptr` has the same end-result, but with more steps (and possibly code) in between, and `return nullptr` depends on an implicit construction. Did I not understand the question?

Reply:
I thought that returning nullptr does not build a shared_ptr, just returns a value and thus more effective. I will check the disassembly

Reply:
`return {}` calls default `shared_ptr` constructor.
`call    std::shared_ptr<structType>::shared_ptr() [complete object constructor]`

`return nullptr` calls constructor for type.
`call    std::shared_ptr<structType>::shared_ptr(decltype(nullptr)) [complete object constructor]`

With optimizations both are optimized away...